src/main/java/uk/ac/cam/cl/dtg/isaac/api/QuizFacade.java

```diff
@@ -0,0 +1,192 @@
+import uk.ac.cam.cl.dtg.segue.dao.SegueDatabaseException;
+import uk.ac.cam.cl.dtg.segue.dao.content.ContentManagerException;
+import uk.ac.cam.cl.dtg.segue.dao.content.IContentManager;
+import uk.ac.cam.cl.dtg.segue.dos.users.Role;
+import uk.ac.cam.cl.dtg.segue.dto.ResultsWrapper;
+import uk.ac.cam.cl.dtg.segue.dto.SegueErrorResponse;
+import uk.ac.cam.cl.dtg.segue.dto.content.ContentSummaryDTO;
+import uk.ac.cam.cl.dtg.segue.dto.users.AbstractSegueUserDTO;
+import uk.ac.cam.cl.dtg.segue.dto.users.RegisteredUserDTO;
+import uk.ac.cam.cl.dtg.util.PropertiesLoader;
+
+import javax.servlet.http.HttpServletRequest;
+import javax.ws.rs.ForbiddenException;
+import javax.ws.rs.GET;
+import javax.ws.rs.Path;
+import javax.ws.rs.PathParam;
+import javax.ws.rs.Produces;
+import javax.ws.rs.core.Context;
+import javax.ws.rs.core.EntityTag;
+import javax.ws.rs.core.MediaType;
+import javax.ws.rs.core.Request;
+import javax.ws.rs.core.Response;
+
+import static uk.ac.cam.cl.dtg.segue.api.Constants.NUMBER_SECONDS_IN_ONE_HOUR;
+
+/**
+ * Games boards Facade.
+ */
+@Path("/quiz")
+@Api(value = "/quiz")
+public class QuizFacade extends AbstractIsaacFacade {
+    private final IContentManager contentManager;
+    private final QuizManager quizManager;
+    private final UserAccountManager userManager;
+
+    private static final Logger log = LoggerFactory.getLogger(QuizFacade.class);
+
+    /**
+     * QuizFacade. For management of quizzes
+     *  @param properties
+     *            - global properties map
+     * @param logManager
+     *            - for managing logs.
+     * @param contentManager
+     *            - for the content etag
+     * @param quizManager
+     *            - for quiz interaction
+     * @param userManager
+     *            - for the currently logged in user
+     */
+    @Inject
+    public QuizFacade(final PropertiesLoader properties, final ILogManager logManager,
+                      final IContentManager contentManager, final QuizManager quizManager,
+                      final UserAccountManager userManager) {
+        super(properties, logManager);
+        this.contentManager = contentManager;
+
+        this.quizManager = quizManager;
+        this.userManager = userManager;
+    }
+
+    /**
+     * Get quizzes visible to this user.
+     *
+     * Anonymous users can't see quizzes.
+     * Students can see quizzes with the visibleToStudents flag set.
+     * Teachers and higher can see all quizzes.
+     *
+     * @return a Response containing a list of ContentSummaryDTO for the visible quizzes.
+     */
+    @GET
+    @Path("available")
+    @Produces(MediaType.APPLICATION_JSON)
+    @GZIP
+    @ApiOperation(value = "Get quizzes visible to this user.")
+    public final Response getAvailableQuizzes(@Context final HttpServletRequest request) {
+        try {
+            AbstractSegueUserDTO user = this.userManager.getCurrentUser(request);
+
+            if (!isUserRegistered(user)) {
```

Comment:
I think the usual pattern for this check is `userManager.getCurrentRegisteredUser(request)` and then to catch the exception that throws for anonymous users:
```java
} catch (NoUserLoggedInException e) {
    return SegueErrorResponse.getNotLoggedInResponse();
}
```
in places where we don't want anonymous users at all. That is probably why there is no `isUserRegistered` method already. Could you swap to that style?

Reply:
See e.g. the [Assignment Facade](https://github.com/isaacphysics/isaac-api/blob/v2.15.2/src/main/java/uk/ac/cam/cl/dtg/isaac/api/AssignmentFacade.java#L254).